evap/results/exporters.py

```diff
@@ -122,53 +133,56 @@ def filter_evaluations(semesters, evaluation_states, degrees, course_types, cont
-                        or question_result.counts is None
-                        or sum(question_result.counts) == 0
-                        for question_result in questionnaire_result.question_results
+                        not isinstance(question_result, RatingResult)
+                        or (question_result.counts is None or sum(question_result.counts) == 0)
+                        for question_result in question_results
                     ):
                         continue
                     if (
                         not contributor
                         or contribution_result.contributor is None
                         or contribution_result.contributor == contributor
                     ):
-                        results.setdefault(questionnaire_result.questionnaire.id, []).extend(
-                            questionnaire_result.question_results
-                        )
+                        results.setdefault(questionnaire_result.questionnaire.id, []).extend(question_results)
                         used_questionnaires.add(questionnaire_result.questionnaire)
-            evaluation.course_evaluations_count = evaluation.course.evaluations.count()
-            if evaluation.course_evaluations_count > 1:
+            monkeypatched_evaluation: Any = evaluation
+            monkeypatched_evaluation.course_evaluations_count = monkeypatched_evaluation.course.evaluations.count()  # type: ignore[attr-defined]
+            if monkeypatched_evaluation.course_evaluations_count > 1:  # type: ignore[attr-defined]
                 course_results_exist = True
-                weight_sum = sum(evaluation.weight for evaluation in evaluation.course.evaluations.all())
-                evaluation.weight_percentage = int((evaluation.weight / weight_sum) * 100)
-                evaluation.course.avg_grade = distribution_to_grade(
-                    calculate_average_course_distribution(evaluation.course)
+                weight_sum = sum(evaluation.weight for evaluation in monkeypatched_evaluation.course.evaluations.all())
+                monkeypatched_evaluation.weight_percentage = int((evaluation.weight / weight_sum) * 100)  # type: ignore[attr-defined]
+                monkeypatched_evaluation.course.avg_grade = distribution_to_grade(  # type: ignore[attr-defined]
+                    calculate_average_course_distribution(monkeypatched_evaluation.course)
                 )
-            evaluations_with_results.append((evaluation, results))
+            evaluations_with_results.append((monkeypatched_evaluation, results))
 
         evaluations_with_results.sort(
             key=lambda cr: (cr[0].course.semester.id, cr[0].course.type.order, cr[0].full_name)
         )
-        used_questionnaires = sorted(used_questionnaires)
 
-        return evaluations_with_results, used_questionnaires, course_results_exist
+        return evaluations_with_results, sorted(used_questionnaires), course_results_exist
```

Comment:
Not related to typing. I liked the old code a bit more. The `sorted` here can easily be overlooked, and before, we had a single block that took care of sorting, now the sorting is more distributed / entagled in the surrounding code.

Reply:
Type checker complains about redefinition

Reply:
I think that doesn't void the argument ;)

Reply:
Suggestion that should satisfy the type checker and us:
```
# [...] other sorting line here
sorted_used_questionnaires = sorted(used_questionnaires)

return evaluations_with_results, sorted_used_questionnaires, course_results_exist
```